h5py/_hl/dataset.py

```diff
@@ -1115,17 +1115,15 @@ def __array__(self, dtype=None, copy=None):
     @with_phil
     def __repr__(self):
         if not self:
-            r = '<Closed HDF5 dataset>'
+            return "<Closed HDF5 dataset>"
+
+        if self.name is None:
+            namestr = "(anonymous)"
         else:
-            if self.name is None:
-                namestr = '("anonymous")'
-            else:
-                name = pp.basename(pp.normpath(self.name))
-                namestr = '"%s"' % (name if name != '' else '/')
-            r = '<HDF5 dataset %s: shape %s, type "%s">' % (
-                namestr, self.shape, self.dtype.str
-            )
-        return r
+            name = pp.basename(pp.normpath(self.name))
+            namestr = f'"{name}"' if name else "/"
```

Comment:
```suggestion
            namestr = f'"{name if name else "/"}"'
```

to make sure we always have quotes in the repr.